backend/coins/btc/handlers/handlers.go

```diff
@@ -364,6 +364,52 @@ func (handlers *Handlers) getAccountBalance(*http.Request) (interface{}, error)
 	}, nil
 }
 
+type slip24Request struct {
+	RecipientName string `json:"recipientName"`
+	Nonce         string `json:"nonce"`
+	Memos         []struct {
+		Type         string `json:"type"`
+		Text         string `json:"text"`
+		Refund       string `json:"refund,omitempty"`
+		CoinPurchase struct {
+			CoinType int    `json:"coinType"`
+			Amount   string `json:"amount,omitempty"`
+			Address  string `json:"address,omitempty"`
+		} `json:"coinPurchase"`
+	} `json:"memos,omitempty"`
+	Outputs []struct {
+		Amount  float64 `json:"amount"`
+		Address string  `json:"address"`
+	} `json:"outputs,omitempty"`
+	Signature string `json:"signature,omitempty"`
+}
+
+func (slip24 slip24Request) toPaymentRequest() (*accounts.PaymentRequest, error) {
+	paymentRequest := &accounts.PaymentRequest{
+		RecipientName: slip24.RecipientName,
+		Nonce:         slip24.Nonce,
+		Signature:     slip24.Signature,
+	}
+
+	for _, memo := range slip24.Memos {
+		if memo.Type != "text" {
+			return nil, errp.New("Payment request non-text memo unsupported")
+		}
+		paymentRequest.Memos = append(paymentRequest.Memos, accounts.TextMemo{Note: memo.Text})
+	}
+
+	if len(slip24.Outputs) != 1 {
+		return nil, errp.New("Missing or multiple payment request output unsupported")
+	}
+	stringAmount := fmt.Sprintf("%.8f", slip24.Outputs[0].Amount)
```

Comment:
Isn't the request-address slip24 amount in sats? Why all of this conversion stuff?

Reply:
Dang, you're right :man_facepalming: I don't know why I thought it was in BTC